subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -214,10 +214,18 @@ uint8_t ll_sync_create_cancel(void **rx)
 	node_rx = (void *)&sync->node_rx_lost;
 	node_rx->hdr.type = NODE_RX_TYPE_SYNC;
 	node_rx->hdr.handle = 0xffff;
-	node_rx->hdr.rx_ftr.param = sync;
+
+	/* NOTE: struct node_rx_lost has uint8_t member following the
+	 *       struct node_rx_hdr to store the reason.
+	 */
 	se = (void *)node_rx->pdu;
 	se->status = BT_HCI_ERR_OP_CANCELLED_BY_HOST;
 
+	/* NOTE: Since NODE_RX_TYPE_SYNC is only generated from ULL context,
+	 *       pass ULL context as parameter.
+	 */
+	node_rx->hdr.rx_ftr.param = sync;
+
 	*rx = node_rx;
```

Comment:
I guess this is equivalent to writing
```
sync->node_rx_lost.hdr.type = NODE_RX_TYPE_SYNC;
sync->node_rx_lost.hdr.handle = 0xffff;
sync->node_rx_lost.hdr.rx_ftr.param = sync;
sync->node_rx_lost.reason = BT_HCI_ERR_OP_CANCELLED_BY_HOST;

*rx = &sync->node_rx_lost;
```
Isn't it? 

I wonder if this is more readable, as you're directly changing the values of the fields in the `sync->node_rx_lost`, rather than doing it through 2 other pointers. 

Similar comment also applies for the `ll_connect_disable` change above.

Reply:
The code is kept similar to HCI access to the reason, this way future code/features can follow the design.

Reply:
I understand that, and will not block the PR based on that. 

However, if using the fields of the `sync->node_rx_lost` it would be possible, if ever needed, to expand that easier than if just expected a certain sequence of fields in that struct. But it's alright as it is right now.